plugins/kubernetes/app/models/kubernetes/role.rb

```diff
@@ -24,13 +24,18 @@ class Role < ActiveRecord::Base
     # create initial roles for a project by reading kubernetes/*{.yml,.yaml,json} files into roles
     def self.seed!(project, git_ref)
       kubernetes_config_files_in_repo(project, git_ref).each do |config_file|
+        scope = where(project: project)
+        next if scope.where(config_file: config_file.file_path).exists?
+
         service_name = config_file.service && config_file.service.metadata.name
-        service_name << "-#{rand(9999999)}" if where(service_name: service_name).exists?
+        service_name << "-#{rand(9999999)}" if service_name && scope.where(service_name: service_name).exists?
```

Comment:
this is happening in the background, so we can either:
 - create an obviously bad name (this PR)
 - fail and block (would block stage/project creation or seeding of other roles ... bad)
 - skip this role ... I think that's bad too since it might go unnoticed ...

maybe we simply do not allow deploying with on of these generated names ... so deploys will fail with an obvious error ...